modyn/evaluator/internal/metric_factory.py

```diff
@@ -1,19 +1,17 @@
-from typing import Any
-
+from modyn.config.schema.pipeline import validate_metric_config_json
 from modyn.evaluator.internal.metrics import AbstractEvaluationMetric, AbstractHolisticMetric, Accuracy, F1Score, RocAuc
 
 all_metrics: set[type[AbstractEvaluationMetric]] = {Accuracy, F1Score, RocAuc}
 
 
 class MetricFactory:
     @staticmethod
-    def get_evaluation_metric(
-        name: str, evaluation_transform_func: str, config: dict[str, Any]
-    ) -> AbstractEvaluationMetric:
+    def get_evaluation_metric(config_json: str) -> AbstractEvaluationMetric:
+        parsed_metric_config = validate_metric_config_json(str(config_json))
```

Comment:
why str()? config_json already is a string

Reply:
I remember there was some issue with the integration tests where the passed value was a JsonString and not a str. I can check again.

Reply:
I guess then we should rather adjust the integration test :D

Reply:
Yeah, I was a bit lazy with debugging😅